src/components/Card.js

```diff
@@ -1,34 +1,55 @@
 class Card {
-  constructor(data, cardSelector, handleCardClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._name = data.name;
     this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDelete = handleDeleteClick;
+    this._handleLike = handleLikeClick;
+    this._likes = Array.isArray(data.likes) ? data.likes : [];
+    this._cardID = data.id;
+  }
+
+  likeCountRemove() {
+    return api
```

Comment:
You should make all the api requests in index.js. The Card should not know anything about the api